superset-frontend/src/explore/actions/saveModalActions.js

```diff
@@ -60,33 +61,132 @@ export function removeSaveModalAlert() {
   return { type: REMOVE_SAVE_MODAL_ALERT };
 }
 
-export function saveSlice(formData, requestParams) {
-  return dispatch => {
-    const url = getExploreUrl({
-      formData,
-      endpointType: 'base',
-      force: false,
-      curUrl: null,
-      requestParams,
+export const getSlicePayload = (sliceName, formData, owners) => {
+  const [datasourceId, datasourceType] = formData.datasource.split('__');
+  const payload = {
+    params: JSON.stringify(formData),
+    slice_name: sliceName,
+    viz_type: formData.viz_type,
+    datasource_id: parseInt(datasourceId, 10),
+    datasource_type: datasourceType,
+    dashboards: formData.dashboards,
+    owners,
+    query_context: JSON.stringify(
+      buildV1ChartDataPayload({
+        formData,
+        force: false,
+        resultFormat: 'json',
+        resultType: 'full',
+        setDataMask: null,
+        ownState: null,
+      }),
+    ),
+  };
+  return payload;
+};
+
+const addToasts = (isNewSlice, sliceName, addedToDashboard) => {
+  const toasts = [];
+  if (isNewSlice) {
+    toasts.push(
+      addSuccessToast(`${t('Chart')} [${sliceName}] ${t('has been saved')}`),
```

Comment:
```suggestion
      addSuccessToast(t('Chart %s has been saved', sliceName)),
```
That's the correct way of handling translations of strings with parameters. We can't use template literals in translations, because it would be impossible to use the params correctly if target language had different sentence structure than English

Reply:
Oops makes sense, thanks!